files/external_node_v2.rb

```diff
@@ -163,6 +165,8 @@ def generate_fact_request(certname, filename)
     rescue => e
       raise "Could not generate facts for Foreman: #{e}"
     end
+  else
+    $stderr.puts 'Last facts are older than last upload time, rerun puppet agent on host to get newer facts'
```

Comment:
You'd get it every time you run node.rb by hand, as the facts file wouldn't have been refreshed.